setup.py

```diff
@@ -1,9 +1,61 @@
 #!/usr/bin/env python
 
-from distutils.core import setup
-from catkin_pkg.python_setup import generate_distutils_setup
```

Comment:
What is gained here by removing the dependency on `catkin_pkg`, which is itself a PyPI package?

Reply:
The main benefit of this is that you do not need to have `catkin_pkg` installed to be able to run the setup.py script to invoke the installation. A standard python package you should be able to naively clone it and then run `python setup.py install` and it will find the declared dependencies install them and then run the build/install steps for the current package. Without this change if you try to follow that standard workflow w/o `catkin_pkg` preinstalled the setup.py will fail before being able to parse the dependencies that says it needs catkin_pkg and you're stuck until you manually determine that catkin_pkg has to be installed and install it yourself, then try to run catkin's setup.py again.

Reply:
Oh I see. So the only things you can really safely import in the `setup.py` context are basically pip, setuptools, and wheel— the stuff that's available in a bare virtualenv.